src/kibali/AuthZChecker.cs

```diff
@@ -82,6 +86,34 @@ public AccessRequestResult CanAccess(string url, string method, string scheme, s
             return AccessRequestResult.InsufficientPermissions;
         }
 
+        public void Validate(PermissionsDocument permissionsDocument)
+        {
+            // Walk permissions, find each pathSet and add path to dictionary
+            foreach (var permission in permissionsDocument.Permissions)
+            {
+                foreach(var pathSet in permission.Value.PathSets)
+                {
+                    foreach (var path in pathSet.Paths)
+                    {
+                        ProtectedResource resource;
+                        if (resources.ContainsKey(path.Key))
+                        {
+                            resource = resources[path.Key];
+
+                        }
+                        else
+                        {
+                            resource = new ProtectedResource(path.Key);
```

Comment:
If the path.Key does not already exist in resources collection then something must have gone wrong in the invertPermissions method and creating a new ProtectedResource here is not going to setup the ProtectedResource properly.  I'm not exactly sure what the right corrective action is.  Perhaps you could emit another error if you find a path that doesn't exist as a protected Resource.

Reply:
Actually, I've merged the two loops so this check is only being done when the resource collection is being built.